mgmt/BaseManager.cc

```diff
@@ -22,35 +22,51 @@
  */
 
 #include "tscore/ink_memory.h"
+#include "tscore/ink_mutex.h"
 #include "BaseManager.h"
 
 BaseManager::BaseManager()
 {
-  /* Setup the event queue and callback tables */
-  mgmt_event_queue = create_queue();
-
-} /* End BaseManager::BaseManager */
+  ink_sem_init(&q_sem, 0);
+}
 
 BaseManager::~BaseManager()
 {
-  while (!queue_is_empty(mgmt_event_queue)) {
-    MgmtMessageHdr *mh = (MgmtMessageHdr *)dequeue(mgmt_event_queue);
-    ats_free(mh);
+  while (!queue.empty()) {
+    ats_free(queue.front());
+    queue.pop();
   }
-  ats_free(mgmt_event_queue);
+}
 
-  for (auto &&it : mgmt_callback_table) {
-    MgmtCallbackList *tmp, *cb_list = it.second;
+bool
```

Comment:
Why isn't this void?

Reply:
I'm not sure. I was following the pattern established in `LLQ`. It should probably be `void` or `self_type`.